tests/integration/test_array_split.py

```diff
@@ -39,11 +40,42 @@
 ]
 
 
+class TestArraySplitErrors:
+    """
+    this class is to test negative cases
+    """
+
+    def test_indices_negative(self):
+        # negative indices should be not accepted
+        ary = np.arange(10)
+        # In numpy, it raises ValueError,
+        # msg is "number sections must be larger than 0."
+        # while in cunumeric, it passed.
+        num.array_split(ary, -2)
+
+    def test_indices_0(self):
+        # 0 indices should be not accepted
+        ary = np.arange(10)
+        # In numpy, it raises ValueError,
+        # msg is "number sections must be larger than 0.".
+        # In cunumeric, it raises ZeroDivisionError
+        with pytest.raises(ZeroDivisionError):
+            num.array_split(ary, 0)
+
+    def test_axis_bigger(self):
+        msg = "has less dimensions than"
+        ary = np.arange(9)
+        with pytest.raises((ValueError, AssertionError), match=msg):
+            num.array_split(ary, len(ary) // 2, 2)
```

Comment:
Numpy raises an `IndexError`  in this case:
```
In [1]: import numpy as np

In [2]: ary = np.arange(9)

In [3]: np.array_split(ary, len(ary) // 2, 2)
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 np.array_split(ary, len(ary) // 2, 2)

File <__array_function__ internals>:180, in array_split(*args, **kwargs)

File ~/anaconda3/envs/gas38/lib/python3.8/site-packages/numpy/lib/shape_base.py:767, in array_split(ary, indices_or_sections, axis)
    741 """
    742 Split an array into multiple sub-arrays.
    743 
   (...)
    764 
    765 """
    766 try:
--> 767     Ntotal = ary.shape[axis]
    768 except AttributeError:
    769     Ntotal = len(ary)

IndexError: tuple index out of range
```

Can we fix the corresponding codepath to replace the assertion and raise the same `IndexError` that numpy does?

Reply:
I fixed it on a new commit by updating its code.